domain/entities/topic_test.go

```diff
@@ -101,13 +126,24 @@ func TestDispatch(t *testing.T) {
 			}
 
 			wg.Wait()
+
 			for _, msg := range tt.messages {
-				topic.Dispatch(msg)
+				err := topic.Dispatch(msg)
+				if tt.wantError == nil {
+					require.NoError(t, err)
+				}
+
+				assertInvalidMessage(t, err, tt.wantError)
```

Comment:
```suggestion
				assert.ErrorIs(t, err, tt.wantError)
```

Reply:
@matheusmosca hey do you like this?

```
func assertDispatchMessages(t *testing.T, topic Topic, messages []vos.Message, wantErr error) {
	for _, msg := range messages {
		err := topic.Dispatch(msg)
		if wantErr == nil {
			require.NoError(t, err)
		}

		assert.ErrorIs(t, err, err)
	}
}
```

the linter warning about: cognitive complexity 11 of func `TestDispatch` is high (> 10) (gocognit) in this approach

Reply:
Sure, but take a look at my new comments @sptGabriel